src/index.ts

```diff
@@ -11,16 +11,28 @@ function valueIsUnchanged(field: HTMLInputElement | HTMLTextAreaElement): boolea
 
 type StorageFilter = (field: HTMLInputElement | HTMLTextAreaElement) => boolean
 
-type PersistOptions = {
+type PersistOptionsWithSelector = {
+  scope?: ParentNode
   selector?: string
+  fields?: never
+}
+
+type PersistOptionsWithFields = {
+  fields?: NodeList | Node[]
+  selector?: never
+}
```

Comment:
```suggestion
type PersistOptionsWithFields = {
  fields?: NodeList | Node[]
  selector?: never
  scope?: never
}
```

[TS Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAChBOBnAlo4B5MxkHsB2iA6ssABYDKEANhAMbA7xQC8UA3gFBTeLV0PwA-AC4oaeMjwBzLj1o5IIscAnTZUAGbJqAE0RK8EAG4IuAXw4dQkWAhRpM2fERKkAYtqp6W7dVt36ogByODoQADKowFAAPlAhYQDaALrqvDT0jAbGptxi8oqihibw5pbW0HBIUY64BD4AFFX2GFh1LmQeAbG21Q5tzsRklBkCAJRQAGS+eQDWECAw8BBaAB5K4pIyeWiMAIZSEEowyLSzADzkAgcQADRQAOSHwACSwBAAtg89D7yv718AHwcCwcDQAVzw9DqUEgfWAACUIIhwR89gAjGhdLyIBrIHSiTbSe4KJwEUTNGoDAgTTh5KAAegZUAAdGyIMBaCDLHCWkiUWjMRBsXoGr8+Jl4A97mx0vxGKIAESK+7zRbLNaiB5qh5mMY8uxRfmojFYzyi34FCDS9iIK1KlVQNVLFbIVZanV6g3w42Cs0BMVyyX5BTWmVBgQO+520NRp0LF2ax6e-UcXlG5EmoUi3EPfw4m1sfN6UQpVUJjVuj0LXWp9NoX2m4Xm3OGVwIQvOyvu5M1r1pw0NzN+5sBh7FBBiCUCQvFxCl5LR6cKqDK8vq1097V9uuDxHDps5sUTpgxyCzlsL6P21eOrub6sgWvevkH7Mt485U-L09Wi8BK8p1GFc1xDSA43vJNtyfftLC4IA)